plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProvider.java

```diff
@@ -103,23 +106,29 @@ public VFSFileProvider( Bowl bowl, KettleVFSService kettleVFSService ) {
     return TYPE;
   }
 
+  private ConnectionManager getConnectionManager() throws MetaStoreException {
+    return bowl.getConnectionManager();
+  }
+
   /**
    * Determines if a <code>filePath</code> is a VFS file
+   *
    * @param filePath
    * @return true if VFS file, false otherwise.
    */
   public boolean isSupported( String filePath ) {
     if ( filePath == null ) {
       return false;
     }
+
     boolean ret = false;
     try {
-      ConnectionUriParser connectionUriParser = new ConnectionUriParser( filePath );
-      String testScheme = connectionUriParser.getScheme() + "://";
-      ret = connectionUriParser.getScheme() != null && testScheme.matches( ConnectionFileSystem.DOMAIN_ROOT );
-    } catch ( PatternSyntaxException e ) {
+      vfsConnectionManagerHelper.parsePvfsUri( filePath );
```

Comment:
**Not a PR merge blocking comment**

I feel that the caller of the function should atleast do a "sanity check" from the results from `vfsConnectionManagerHelper.parsePvfsUri(...)`.

like
`ConnectionFIleName != null && ConnectionFIleName.getURI() == filepath`

I get that you are relying on a `KettleException` being thrown if the parsing fails.

But what if the implementation of the `vfsConnectionManagerHelper#parsePvfsUri` changes and the null return value signifies a failed parsing.

Reply:
I am still to write the unit tests for `VFSConnectionManagerHelper.parsePvfsUri(.)`.

This method is properly documented, and annotated with `@NonNull`, and has checked exceptions:
```java
/**
   * Parses a given PVFS URI string.
   * @param pvfsUri The PVFS URI string
   * @return The connection file name.
   * @throws KettleException When the given PVFS URI is not a valid PVFS URI.
   */
  @NonNull
  public ConnectionFileName parsePvfsUri( @NonNull String pvfsUri ) throws KettleException {
```

If this ever changes, it is a clear breaking change, responsibility of who makes it!
Not sure it is a good idea to be "defense-programming" with APIs which are well defined/documented and are maintained by ourselves.
I mean, I would need to add many assertions to code being proposed.

Reply:
Rergarding the previous comment. Got rid of `VFSConnectionManagerHelper.parsePvfsUri(.)`, in favor of directly calling `ConnectionFileParser#parseUri(.)`, whose signature and documentation is the following:

```java
/**
   * Parses a PVFS URI.
   *
   * @param pvfsUri The PVFS URI to parse.
   * @return The corresponding connection file name.
   * @throws FileSystemException When the given PVFS URI is not valid.
   */
  @NonNull
  public ConnectionFileName parseUri( @NonNull String pvfsUri ) throws FileSystemException {
    Objects.requireNonNull( pvfsUri );
```

I believe my previous arguments still stand.